### ISSUE TITLE
Can we make HttpClientFactory as purely configuration API?

### ISSUE THREAD

**Issue by SamGuoMsft** [dotnet/runtime#74508]
Starts with Dot Net Core 2.1, the team introduced SocketsHttpHandler which included a connection pooling system and the ability to set maximum lifetimes for those connections to solve DNS change issue. This is redundant with one of the features of IHttpClientFactory is the lifetime management of HttpMessageHandler chains and as a result, also of the underlying connections.

Based on the [post ](https://github.com/dotnet/aspnetcore/issues/28385#issuecomment-661944137) from @davidfowl:

_**The HttpClientFactory is meant to be a configuration API, not a lifetime management API. In fact, we're likely going to remove the handler pooling and instead set the appropriate settings based on the platform to avoid this confusion**_

Since it has been years after the post above, would like to know whether we really have plan to make changes on the HttpClientFactory since I see more and more libraries(e.g.: grpc client factory) are based on HttpClientFactory since I am big fan about the configuration flexibility that HttpClientFactory provided. 

The other issue which I want to bring up is with type client model since the typed service is registered with transient scope. This works in cases where you plan to consume the typed service from another transient service. A common place to use these in ASP.NET Core will be places such as Controllers for example. That works as expected since the controller is created by the framework for each request.
However, what if you want to use the typed client within a singleton service? That presents a problem. If this was allowed it would be created and injected once, and then held onto by the singleton service. This is not the behavior we want.
There are approaches to solve the issue which I can think:
1.	Inject HttpClientFactory and use name model instead of type model.
2.	Build a factory model (more details are from [IHttpClientFactory Patterns: Using Typed Clients from Singleton Services - Steve Gordon - Code with Steve (stevejgordon.co.uk)](https://www.stevejgordon.co.uk/ihttpclientfactory-patterns-using-typed-clients-from-singleton-services)
3.	Replace the PrimaryHttpMessageHandler with SocketsHttpHandler with PooledConnectionLifeTime, so that we can use the injected httpclient which created from HttpClientFactory as singleton. But this is weird since we use HttpClientFactorty handler pool to create handler chain with primary handler as SocketsHttpHanler

```c#
.ConfigurePrimaryHttpMessageHandler(() =>
{
    return new SocketsHttpHandler
    {
        PooledConnectionLifetime = TimeSpan.FromSeconds(200)
    };
})
```

**Comment by CarnaViire** [comment]
Thanks for the suggestions @SamGuoMsft.

I see that there are two main points:

1) How to stop `HttpClientFactory` from managing handler's lifetime and use `SocketsHttpHandler` connection pooling instead
2) How to safely use typed clients in singleton services

This can currently be achieved by setting `SocketsHttpHandler` as a primary handler (as you mentioned) and at the same time turning off factory's handler rotation by setting `HandlerLifetime` to `Timeout.InfiniteTimeSpan`.
(Linked question: #74173)

------

There are still (future) plans to make changes to `HttpClientFactory` to turn off handler rotation and utilize `SocketsHttpHandler` by default. Tracking issue is #35987. Note that `HttpClientFactory` is also available on platforms that don't support `SocketsHttpHandler`.

As for Typed client usage in singletons, there're additional considerations tracked in https://github.com/dotnet/runtime/issues/64034#issuecomment-1020581001, which will most likely depend on #47091

I believe these two issues should cover both your points, so I will close this issue as a duplicate. Please feel free to reopen if I missed something. Thanks!

**Comment by CarnaViire** [comment]
Duplicate of #35987
